File: bus/src/main/java/com/iwdael/livedatabus/ObservableWrapper.java
package com.iwdael.livedatabus;

import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.MutableBusLiveData;
import androidx.lifecycle.Observer;

public class ObservableWrapper<T> implements Observable<T> {
    final MutableBusLiveData<T> liveData;

    public ObservableWrapper(MutableBusLiveData<T> liveData) {
        this.liveData = liveData;
    }

    @Override
    public void observe(@NonNull LifecycleOwner owner, @NonNull Observer<? super T> observer) {
        this.liveData.observe(owner, observer);
    }

    @Override
    public void observeSticky(@NonNull LifecycleOwner owner, @NonNull Observer<? super T> observer) {
        this.liveData.observeSticky(owner, observer);
    }

    @Override
    public void observeForever(@NonNull Observer<? super T> observer) {
        this.liveData.observeForever(observer);
    }

    @Override
    public void observeForeverSticky(@NonNull Observer<? super T> observer) {
        this.liveData.observeForeverSticky(observer);
    }

    @Override
    public void removeObserver(@NonNull Observer<? super T> observer) {
        this.liveData.removeObserver(observer);
    }
}

File: core/src/main/java/com/iwdael/livedatabus/ObservableWrapper.kt
package com.iwdael.livedatabus

import androidx.lifecycle.LifecycleOwner
import com.iwdael.livedatabus.ObservableWrapper
import androidx.lifecycle.MutableBusLiveData
import androidx.lifecycle.Observer
/**
 * author : Iwdael
 * e-mail : iwdael@outlook.com
 */
class ObservableWrapper<T>(val liveData: MutableBusLiveData<T>) : Observable<T> {
    override fun observe(owner: LifecycleOwner, observer: Observer<in T>) {
        liveData.observe(owner, observer)
    }

    override fun observeSticky(owner: LifecycleOwner, observer: Observer<in T>) {
        liveData.observeSticky(owner, observer)
    }

    override fun observeForever(observer: Observer<in T>) {
        liveData.observeForever(observer)
    }

    override fun observeForeverSticky(observer: Observer<in T>) {
        liveData.observeForeverSticky(observer)
    }

    override fun removeObserver(observer: Observer<in T>) {
        liveData.removeObserver(observer)
    }
}